src/SSHDebugPS/CommandRunner.cs

```diff
@@ -9,304 +9,304 @@
-    internal class LocalSingleCommandRunner : ICommandRunner
+    internal class LocalCommandRunner : ICommandRunner
     {
-        private System.Diagnostics.Process _localProcess;
-        private StreamWriter _stdoutWriter;
-        private StreamReader _processReader;
+        public static int BUFMAX = 4096;
+
+        private ProcessStartInfo _processStartInfo;
+        private System.Diagnostics.Process _process;
         private CancellationTokenSource _cancellationSource = new CancellationTokenSource();
-        private bool _isClosed = false;
-        private ProcessStartInfo processStartInfo;
+        private System.Threading.Tasks.Task _outputReadLoopTask;
+        private System.Threading.Tasks.Task _errorReadLoopTask;
         private bool _hasExited = false;
-        private object _lock = new object();
-        private System.Threading.Tasks.Task _outputReadTask = null;
 
-        public LocalSingleCommandRunner(string command, string arguments)
-        {
-            processStartInfo = new ProcessStartInfo(command, arguments);
-            processStartInfo.RedirectStandardError = true;
-            processStartInfo.RedirectStandardInput = true;
-            processStartInfo.RedirectStandardOutput = true;
-            processStartInfo.UseShellExecute = false;
-            processStartInfo.CreateNoWindow = true;
-        }
+        private StreamWriter _stdInWriter;
+        private StreamReader _stdOutReader;
+        private StreamReader _stdErrReader;
+
+        protected object _lock = new object();
+
+        public event EventHandler<string> OutputReceived;
+        public event EventHandler<int> Closed;
+        public event EventHandler<ErrorOccuredEventArgs> ErrorOccured;
+
+        public LocalCommandRunner(IPipeTransportSettings settings)
+            : this(settings.Command, settings.CommandArgs)
+        { }
 
-        public void Run()
+        public LocalCommandRunner(string command, string commandArgs)
         {
-            _localProcess = new System.Diagnostics.Process();
-            _localProcess.StartInfo = processStartInfo;
-            _localProcess.Exited += OnProcessExited;
-            _localProcess.EnableRaisingEvents = true;
-            _localProcess.Start();
-
-            _stdoutWriter = new StreamWriter(_localProcess.StandardInput.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), 4096, leaveOpen: false);
-            _processReader = new StreamReader(_localProcess.StandardOutput.BaseStream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), false, BUFMAX, false);
-            _outputReadTask = System.Threading.Tasks.Task.Run(() => ReadLoop(_processReader, _cancellationSource.Token, (line) => { OutputReceived?.Invoke(this, line); }));
-
-            _localProcess.ErrorDataReceived += OnErrorOutput;
-            _localProcess.Exited += OnProcessExited;
-            _localProcess.BeginErrorReadLine();
+            CreateProcessStartInfo(command, commandArgs);
```

Comment:
>CreateProcessStartInfo [](start = 12, length = 22)

Should not call a virtual method from the constructor of a non-sealed class.  In this case, I don't think the method needs to be virtual anyway, so you can either make it non-virtual or just inline it into the constructor.